src/atomics/sys/atomic_stdc.h

```diff
@@ -73,11 +75,11 @@ static inline void pmix_atomic_wmb(void)
 
 static inline void pmix_atomic_rmb(void)
 {
-#    if PMIX_ASSEMBLY_ARCH == PMIX_X86_64
+#    if defined(PMIX_ATOMIC_X86_64)
     /* work around a bug in older gcc versions (observed in gcc 6.x)
      * where acquire seems to get treated as a no-op instead of being
      * equivalent to __asm__ __volatile__("": : :"memory") on x86_64 */
-    pmix_atomic_mb();
+    __asm__ __volatile__("" : : : "memory");
```

Comment:
This might actually be a problem.  Curious to see what IBM tests think of this.  Also not sure how to fix it if they get angry.